### ISSUE TITLE
SRON plume retrieval issue

### ISSUE THREAD

**Issue by nicholasbalasus** [geoschem/integrated_methane_inversion#228]
### Name and Institution (Required)

Name: Nick Balasus
Institution: Harvard University

### Description of your issue or question

Running the IMI dev branch (as of commit e18648e) errors on the plume visualization out of the box (20180501-20180508).

```
Using cached plume data for SRON
/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py:391: UserWarning: Cannot open cached file at /n/holyscratch01/jacob_lab/nbalasus/Test_IMI_Permian_dev/SRON_plumes/plumes.geojson, fetching new data.
  warnings.warn(msg)
Fetching plumes from SRON...
Traceback (most recent call last):
  File "/var/slurmd/spool/slurmd/job37032535/slurm_script", line 679, in <module>
    imi_preview(
  File "/var/slurmd/spool/slurmd/job37032535/slurm_script", line 242, in imi_preview
    point_sources=get_point_source_coordinates(config),
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/point_sources.py", line 79, in get_point_source_coordinates
    observers.append( observer(config, usecached = True) )
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 450, in __init__
    super().__init__(self.myname, config, usecached, self.cache)
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 398, in __init__
    self._cache_data()
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 417, in _cache_data
    self.gdf.to_file(
  File "/n/home06/nbalasus/micromamba/envs/imi_env/lib/python3.9/site-packages/geopandas/geodataframe.py", line 1114, in to_file
    _to_file(self, filename, driver, schema, index, **kwargs)
  File "/n/home06/nbalasus/micromamba/envs/imi_env/lib/python3.9/site-packages/geopandas/io/file.py", line 367, in _to_file
    schema = infer_schema(df)
  File "/n/home06/nbalasus/micromamba/envs/imi_env/lib/python3.9/site-packages/geopandas/io/file.py", line 428, in infer_schema
    raise ValueError("Cannot write empty DataFrame to file.")
ValueError: Cannot write empty DataFrame to file.
```

Similar but different if I run out of the box for 20230401-20230408 (but now for CarbonMapper).
```
Using cached plume data for SRON
/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py:391: UserWarning: Cannot open cached file at /n/holyscratch01/jacob_lab/nbalasus/Test_IMI_Permian_dev/SRON_plumes/plumes.geojson, fetching new data.
  warnings.warn(msg)
Fetching plumes from SRON...
No CarbonMapper points in ROI
No CarbonMapper points in ROI
Traceback (most recent call last):
  File "/var/slurmd/spool/slurmd/job37046555/slurm_script", line 679, in <module>
    imi_preview(
  File "/var/slurmd/spool/slurmd/job37046555/slurm_script", line 242, in imi_preview
    point_sources=get_point_source_coordinates(config),
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/point_sources.py", line 82, in get_point_source_coordinates
    ps = PointSources(gf, observers)
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 62, in __init__
    self.grid_ds = self._grid_all_data()
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 171, in _grid_all_data
    ds_i = self._grid_datasource(d)
  File "/n/home06/nbalasus/test-imi-dev/integrated_methane_inversion/src/inversion_scripts/plumes.py", line 101, in _grid_datasource
    if inout.sum() > 0:
AttributeError: 'NoneType' object has no attribute 'sum'
```

@eastjames maybe there is some missing logic for when no plumes are found.

**Comment by laestrada** [comment]
@eastjames I also ran into this and just decided to wrap the relevant lines in a try except like so:

```
            try:
                observers = []
                for plume_dataset in ps_datasets:
                    # instantiate plume observer object
                    # using its string name from config file
                    observer = getattr(imiplumes, plume_dataset)
                    observers.append( observer(config, usecached = True) )
                    
                gf = GeoFilter(config)
                ps = PointSources(gf, observers)
                plumes += ps.get_gridded_coords(
                    emission_rate_filter = int(config['EmissionRateFilter']),
                    plume_count_filter = int(config['PlumeCountFilter'])
                ) 
                msg = (
                    f'Found {len(plumes)} grid cells with plumes '
                    f'using {[ob.myname for ob in observers]}'
                )
                print(msg, flush=True)
                got_plumes = True
            except:
                print("An unexpected error occurred when fetching plumes." 
                      "Continuing without plume datasets.")
                got_plumes = False
                plumes = []
```

This is probably not ideal though.

**Comment by eastjames** [comment]
Thanks @nicholasbalasus and @laestrada. Nick, could you send me the config file that generated this?

**Comment by eastjames** [comment]
Thanks @nicholasbalasus and @laestrada. I'll work on logic for missing data.

**Comment by eastjames** [comment]
@nicholasbalasus Just to confirm -- this was run out of the box? CarbonMapper is not included in the point source datasets out of the box. The error may not be from CarbonMapper.

**Comment by nicholasbalasus** [comment]
Hi James - out of the box for dev as of https://github.com/geoschem/integrated_methane_inversion/commit/e18648e2f36f8ca00fdb57c047d7344843713490.

**Comment by eastjames** [comment]
Resolved by #229
